### PR TITLE
fix(dashpay): show user-facing identity ids in base58, not hex

### PR DESCRIPTION
**Pull request by UdjinM6**
## Summary

A Dash Platform identity id is 32 raw bytes; the string form is a presentation choice. Everywhere a user meets an identity id it should be **base58** — that is what Platform explorers use, and what the app already reached for whenever it had a dedicated property for the job (`ContactItem.identityIdBase58`, the row model's `idBase58`, the `dashpay://user` QR payload). Hex belongs in logs, cache keys and interop payloads.

The gap was in the places that formatted an id inline instead of using those properties — including two inside the contacts and identities screens that otherwise render base58.

Four user-facing surfaces were still rendering hex, so the id shown in the app could not be compared against an explorer without decoding it by hand. This PR moves all four to base58 and leaves every non-display use of hex alone.

## What changed

| Surface | Before | After |
|---|---|---|
| `SDKIdentityProfileSheet` — "Identity ID" row | 64-char hex | base58 |
| `ShieldedActivityHistory` — `identityCreate` detail "Identity ID" row | 64-char hex | base58 |
| `ContactItem.displayTitle` — last-resort fallback | 8-char hex prefix | 8-char base58 prefix |
| `IdentitiesViewModel` — unnamed identity row title | 12-char hex prefix | 12-char base58 prefix |

Both prefix lengths are unchanged. base58 carries more entropy per character than hex, so the truncations stay at least as collision-resistant.

## Why these were hex

The profile sheet said so in its own doc comment: it rendered hex "matching the existing coordinator logs". The format was inherited from log output rather than chosen for the reader. `IdentitiesViewModel` was internally inconsistent — the row model already carried `idBase58` for the full id while the title fell back to `identityIdString` (hex). `ContactItem` built its own hex prefix four lines above an `identityIdBase58` property whose comment already stated base58 is "the identity id as users see it elsewhere".

## Deliberately unchanged

- Logs (`DWIdentityRegistrationCoordinator`, contacts service)
- UserDefaults / cache keys (`DWCurrentUserIdentityInfo`, `SwiftDashSDKContactsService`)
- `Identifiable` conformances (`ContactCandidate.id`)
- The raw Storage Explorer — left untouched pending a decision from whoever owns that screen

`DWCurrentUserIdentityInfo.identityIdHex` (`@objc`) lost its only caller when the profile sheet moved to base58, so the second commit removes the property, its `Snapshot` field and both construction sites — nothing in the app read it, in Swift or Obj-C. The `hex` local in `rebuildSnapshot()` stays, since the snapshot log line still prints its first 8 characters.

One follow-up is left out of scope. In the Storage Explorer, most rows label the format explicitly (`ID (Hex)`, `Owner ID (Hex)`, `Contact ID (Hex)`) while `Identity ID`, `From`, `To` and `Performed By` render hex without that qualifier. We assumed hex is deliberate there — it is a raw storage inspector — but did not verify that intent, so nothing was changed. Worth a look from whoever owns the screen: either the labels should be consistent, or those rows belong in base58 like the rest of the app.

## Verification

Built clean against the `dashpay` scheme (`ARCHS=arm64`, iOS 26.5 simulator); each touched file confirmed recompiled with no new warnings.

Verified on device:

- **Profile sheet** — the id matches the Identities list and resolves on Platform Explorer.
- **Shielded `identityCreate` row** — displayed `AY4heGwy…4RxvDfTS`; the copied value is 44 characters, every character inside the base58 alphabet, decoding to exactly 32 bytes and round-tripping cleanly. No version byte, no checksum — plain base58, matching `Data.toBase58String()`.

Not exercised at runtime: the two truncated fallbacks, which require a contact with no alias/display name/username and an identity with no name at all. Both are build-verified only.

## Summary by CodeRabbit

* **Improvements**
  * Identity IDs are now consistently displayed in Base58 format across contacts, profiles, identity lists, and shielded activity details.
  * Identity IDs shown in activity details remain shortened for readability and can still be copied.
  * Identity information continues to support avatar and top-up functionality.

### CODE REVIEW

**Comment by thepastaclaw** [comment]
✅ Final review complete — no blockers (commit 75c5c8b8)

**Review by thepastaclaw — COMMENTED** [review]
## Final validation — GLM Flash + Sol

The exact-head changes consistently use the established plain-Base58 helpers across all four intended display surfaces, preserve hexadecimal formatting for non-display uses, and remove an Objective-C accessor with no remaining callers. The two fallback branches that define part of this bug fix still lack focused regression tests, so this review leaves one non-blocking test-coverage suggestion.

Source: reviewer 1: `glm-5.3-flash` (agent: `phase1-reviewer`, role: `general`); reviewer 2: `gpt-5.6-sol` (agent: `phase2-reviewer`, role: `general`); final verifier: `gpt-5.6-sol` (agent: `sol-verifier`, role: `final-verifier`)

### Review provenance
- Phase 1 reviewers (GLM Flash): `glm-5.3-flash` — general (completed); agent `phase1-reviewer`
- Fresh verifier (Sol): `gpt-5.6-sol` — final-verifier; agent `sol-verifier`
- Phase 2 reviewers (Sol): `gpt-5.6-sol` — general (completed); agent `phase2-reviewer`

🟡 1 suggestion(s)

<details>
<summary>🤖 Prompt for all review comments with AI agents</summary>

```
These findings are from an automated code review. Verify each finding against the current code and only fix it if needed.

In `DashWallet/Sources/Infrastructure/SwiftDashSDK/Contacts/ContactItem.swift`:
- [SUGGESTION] DashWallet/Sources/Infrastructure/SwiftDashSDK/Contacts/ContactItem.swift:122: Add regression coverage for the anonymous identity fallbacks
  This contact fallback and the unnamed-identity fallback in `IdentitiesViewModel.swift:359` are two of the four user-facing behaviors fixed by this PR, but neither has compile-ready regression coverage or was exercised at runtime. The existing `DashPayUserLinkTests` verify that the Base58 codec round-trips; they would not catch either display path reverting to hexadecimal. Add focused tests using a fixed 32-byte ID with all naming fields empty and assert the exact Base58-prefixed titles. A leading-zero fixture would clearly distinguish Base58's leading `1` representation from hexadecimal.
```
</details>

File: DashWallet/Sources/Infrastructure/SwiftDashSDK/Contacts/ContactItem.swift [hunk 1]
```diff
@@ -119,12 +119,12 @@ struct ContactItem: Identifiable, Equatable {
         if let username, !username.isEmpty {
             return username.withoutDashSuffix
         }
-        return String(contactIdentityId.map { String(format: "%02x", $0) }.joined().prefix(8)) + "…"
+        return String(identityIdBase58.prefix(8)) + "…"
```
Comment (thepastaclaw) [inline DashWallet/Sources/Infrastructure/SwiftDashSDK/Contacts/ContactItem.swift:122]: **🟡 Suggestion: Add regression coverage for the anonymous identity fallbacks**

This contact fallback and the unnamed-identity fallback in `IdentitiesViewModel.swift:359` are two of the four user-facing behaviors fixed by this PR, but neither has compile-ready regression coverage or was exercised at runtime. The existing `DashPayUserLinkTests` verify that the Base58 codec round-trips; they would not catch either display path reverting to hexadecimal. Add focused tests using a fixed 32-byte ID with all naming fields empty and assert the exact Base58-prefixed titles. A leading-zero fixture would clearly distinguish Base58's leading `1` representation from hexadecimal.

<sub>source: ['claude']</sub>

File: DashWallet/Sources/UI/DashPay/Profile/SDKIdentityProfileSheet.swift [hunk 2]
```diff
@@ -14,7 +14,7 @@ import DashUIKit
 
 struct SDKIdentityProfileSheet: View {
     @Environment(\.dismiss) private var dismiss
-    @State private var identityIdHex: String? = nil
+    @State private var identityIdBase58: String? = nil
```
[the remark this answers is not in the document]
Comment (thepastaclaw) [reply DashWallet/Sources/UI/DashPay/Profile/SDKIdentityProfileSheet.swift:17]: The explicit `= nil` initializer was already present on this `@State` declaration at the merge base; this PR only renames the property. This is a pre-existing, behavior-neutral SwiftLint style warning rather than an issue introduced by the change.
